--- build_periodic/globals.h.orig	2024-07-16 13:33:44.415882713 +0200
+++ build_periodic/globals.h	2024-07-16 13:33:57.052013513 +0200
@@ -32,7 +32,7 @@
       ! hard limits on maximum number of particles: all & sph
       
       LOGICAL periodic
-      PARAMETER (periodic=.FALSE.)
+      PARAMETER (periodic=.TRUE.)
       ! periodic boundary conditions
             
       LOGICAL usepm
