--- src/runtime/include/SharedMemory.h  2019-01-18 14:03:01.005784236 +0100
+++ src/runtime/include/SharedMemory.h  2019-01-18 14:03:01.005784236 +0100
@@ -153,7 +153,7 @@
     using Header    = typename p::Header;
 
   public:
-    SharedMemoryClient(const std::string &descriptor, const double timeOut = HUGE) : p(descriptor) 
+    SharedMemoryClient(const std::string &descriptor, const double timeOut = 10e10f) : p(descriptor) 
     {
       double dt = 0.0;
       while (p::shmfd < 0 && dt < timeOut)


