From 03d46d1aac9c4d4f36a1f3fc86cc7efea7d49d63 Mon Sep 17 00:00:00 2001
From: Rob Farmer <robert.j.farmer37@gmail.com>
Date: Tue, 2 Nov 2021 15:59:01 +0100
Subject: [PATCH] Add mpi error handling

---
 utils/public/utils_lib.f90 | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/utils/public/utils_lib.f90 b/utils/public/utils_lib.f90
index eb5f0fcf..a01e36d9 100644
--- a/utils/public/utils_lib.f90
+++ b/utils/public/utils_lib.f90
@@ -1135,10 +1135,13 @@ contains
 
 
    subroutine mesa_error(file, line,msg)
+      use mpi
       use iso_fortran_env, only: error_unit
       character(len=*), intent(in) :: file
       character(len=*), optional,intent(in) :: msg
       integer, intent(in) :: line
+      integer :: parent, ierr
+      integer, dimension(8) :: header
       character (len=strlen) :: bt_disable
       if(present(msg)) then
          write(error_unit,"(A, A, A, I4, A, A)") "File: ", file, ", Line: ",line,", Message: ",msg
@@ -1146,11 +1149,19 @@ contains
          write(error_unit,"(A, A, A, I4)") "File: ", file, ", Line: ", line
       end if
       call get_environment_variable("MESA_ERROR_BACKTRACE_DISABLE", bt_disable)
-      if (len_trim(bt_disable) > 0) then
-         stop 1
-      else
-         error stop 1
-      end if
+      call MPI_Comm_get_parent(parent, ierr)
+      header = 0
+      header(1) = -2
+      header(2) = 1
+      call MPI_Send(header, 8, MPI_INTEGER, 0, 999, parent, ierr)
+      call MPI_Comm_disconnect(parent, ierr)
+      call MPI_Finalize(ierr)
+      ! if (len_trim(bt_disable) > 0) then
+      !    stop 1
+      ! else
+      !    error stop 1
+      ! end if
+
    end subroutine mesa_error
 
 
-- 
2.31.1

