--- a/fpx3_deps	2021-04-13 11:35:49.104671365 +0200
+++ b/fpx3_deps	2021-04-13 11:27:18.972314103 +0200
@@ -13,23 +13,13 @@
     'iso_fortran_env',
     'iso_c_binding',
     'omp_lib',
-    'omp_lib_kinds',
-    'ieee_arithmetic',
+'hdf5','f95_lapack',    'omp_lib_kinds',
+'hdf5','f95_lapack',    'ieee_arithmetic',
     'ieee_exceptions',
     'ieee_features'
     );
 
-$EXCLUDE_PATHS="$ENV{'@SDK_ROOT@'}/include";
-
-if (exists $ENV{'@SDK_MATH_SLOT@'}) {
-    $EXCLUDE_PATHS .= ":$ENV{'@SDK_ROOT@'}/math-slots/$ENV{'@SDK_MATH_SLOT@'}/include";
-}
-else {
-    $EXCLUDE_PATHS .= ":$ENV{'@SDK_ROOT@'}/math-slots/default/include";
-}
-    
-# Split the command-line arguments into program 
-# units and fpx3 arguments
+$EXCLUDE_PATHS="";# units and fpx3 arguments
 
 @units = grep !/^-/, @ARGV;
 @args = grep /^-/, @ARGV;
